lib/easy-stackup.js

```diff
@@ -0,0 +1,76 @@
+// simpler API main function
+'use strict'
+var gerberToSvg = require('gerber-to-svg')
+var shortId = require('shortid')
+var whatsThatGerber = require('whats-that-gerber')
+
+var pcbStackup = require('./index')
+
+module.exports = function(layers, optionsOrCallback, done) {
+  var options = {}
+
+  if (typeof optionsOrCallback === 'object') {
+    options = optionsOrCallback
+  }
+  else if (typeof optionsOrCallback === 'function') {
+    done = optionsOrCallback
+  }
+
+  if (!('id' in options)) {
+    options.id = shortId.generate()
+  }
+
+  var layerCount = layers.length
+  var stackupLayers = []
+
+  var finishLayer = function() {
+    if (--layerCount < 1) {
+      var stackup = pcbStackup(stackupLayers, options)
+
+      stackup.layers = stackupLayers.map(function(layer) {
+        return {
+          layerType: layer.type.id,
+          gerber: layer.converter,
+          options: layer.options
+        }
+      })
+
+      return done(null, stackup)
+    }
+  }
+
+  if (layerCount === 0) {
+    return finishLayer()
+  }
+
+  for (var i in layers) {
+    var layer = layers[i]
+    var layerType
+    var layerOptions = layer.options || {}
+
+    if ('layerType' in layer) {
+      layerType = layer.layerType
+    }
+    else if ('filename' in layer) {
+      layerType = whatsThatGerber(layer.filename)
+    }
+    else {
+      return done(new Error('No filename or layerType given for layer ' + i))
```

Comment:
This is my only nitpick with all of your comments. I think silently ignoring invalid inputs is a terrible quality in a library. If it were my project I definitely wouldn't allow it and I would go further and try and check all the inputs.

Some poor soul is going to spend a lot of time thinking they have set everything up correctly but it isn't working. It can waste a lot of time and be very frustrating.